homeassistant/components/hydrawise/config_flow.py

```diff
@@ -21,26 +23,62 @@
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Hydrawise."""
 
-    VERSION = 1
+    VERSION = 2
 
-    async def _create_entry(
-        self, api_key: str, *, on_failure: Callable[[str], FlowResult]
+    def __init__(self) -> None:
+        """Construct a ConfigFlow."""
+        self.reauth_entry: ConfigEntry | None = None
```

Comment:
Can we maybe move the reauth to a preliminary PR? This way we keep this PR better scoped

Reply:
Not sure I understand. Are you suggesting to add the reauth path for the old api-key authentication method first?

I would prefer not to do that. Not only would I have to write reauth tests that I would have to immediately discard again in this PR, it also wouldn't make this PR any easier to understand. The reauth path is affected by the api_key to username/password transition and would have to undergo substantial changes again for this PR.

Reply:
Agreed, reauth should stay.